google-api-client/src/main/java/com/google/api/client/googleapis/apache/v2/GoogleApacheHttpTransport.java

```diff
@@ -39,11 +40,33 @@
 public final class GoogleApacheHttpTransport {
 
   /**
-   * Returns a new instance of {@link ApacheHttpTransport} that uses
-   * {@link GoogleUtils#getCertificateTrustStore()} for the trusted certificates.
+   * Returns a new instance of {@link ApacheHttpTransport} that uses {@link
+   * GoogleUtils#getCertificateTrustStore()} for the trusted certificates. If
+   * `GOOGLE_API_USE_CLIENT_CERTIFICATE` environment variable is set to "true", and the default
+   * client certificate key store from {@link Utils#loadDefaultMtlsKeyStore()} is not null, then the
+   * transport uses the default client certificate and is mutual TLS.
    */
-  public static ApacheHttpTransport newTrustedTransport() throws GeneralSecurityException,
-      IOException {
+  public static ApacheHttpTransport newTrustedTransport()
+      throws GeneralSecurityException, IOException {
+    return newTrustedTransport(null, "");
+  }
+
+  /**
+   * {@link Beta} <br/>
+   * Returns a new instance of {@link ApacheHttpTransport} that uses {@link
+   * GoogleUtils#getCertificateTrustStore()} for the trusted certificates. If
+   * `GOOGLE_API_USE_CLIENT_CERTIFICATE` environment variable is set to "true", the function uses
+   * the provided mtlsKeyStore or the default key store from {@link Utils#loadDefaultMtlsKeyStore()}
+   * to create the transport. If either key store exists, then the created transport is mutual TLS.
+   * The provided key store takes precedence over the default one.
+   *
+   * @param mtlsKeyStore KeyStore for mutual TLS client certificate and private key
+   * @param mtlsKeyStorePassword KeyStore password
+   */
+  @Beta
+  public static ApacheHttpTransport newTrustedTransport(
+      KeyStore mtlsKeyStore, String mtlsKeyStorePassword)
+      throws GeneralSecurityException, IOException {
```

Comment:
same here. 

This override is not intended to be general purpose. We need this override to control whether the keystore should be used for mtls, even if it is explicitly provided by the user. Python/Go has a similar override for the same functionality. We will probably need this control for a while until all mtls endpoints are GA. For googleapis, users should use this override in case they want their own certificate (even though this is discouraged).